### PR TITLE
icewm: 1.4.2 -> 1.6.0

### PR DESCRIPTION
**Pull request by volth**
###### Motivation for this change

Version update.

I removed `musl` patch because I was unable to build `pkgsMusl.icewm`, about six of dependencies are broken.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @dtzWill @AndersonTorres

### CODE REVIEW

**Comment by volth** [comment]
@GrahamcOfBorg build icewm
@GrahamcOfBorg build pkgsCross.armv7l-hf-multiplatform.icewm
@GrahamcOfBorg test xrdp

**Comment by volth** [comment]
@GrahamcOfBorg build icewm
@GrahamcOfBorg build pkgsCross.armv7l-hf-multiplatform.icewm
@GrahamcOfBorg test xrdp

File: pkgs/applications/window-managers/icewm/default.nix [hunk 1]
```diff
@@ -1,36 +1,40 @@
...
+    repo   = "icewm";
+    rev    = version;
+    sha256 = "1l8hjmb19d7ds7z21cx207h86wkjcmmmamcnalgkwh4alvbawc2p";
+  };
+
+  nativeBuildInputs = [ cmake perl asciidoc ];
+
+  buildInputs = [
+    gettext libjpeg libtiff libungif libpng imlib expat
     freetype fontconfig pkgconfig gdk-pixbuf mkfontdir libX11
```
Comment (jonringer) [inline pkgs/applications/window-managers/icewm/default.nix]: i think pkgconfig can be moved to native inputs as well.
```suggestion
    freetype fontconfig gdk-pixbuf mkfontdir libX11
```

**Comment by volth** [comment]
@GrahamcOfBorg build icewm
@GrahamcOfBorg build pkgsCross.armv7l-hf-multiplatform.icewm
@GrahamcOfBorg test xrdp

**Comment by AndersonTorres** [comment]
Looks good to me!

**Pull request merged**